src/Chapter2.hs

```diff
@@ -841,7 +878,10 @@ list.
 
 🕯 HINT: Use the 'cycle' function
 -}
-rotate = error "rotate: Not implemented!"
+rotate :: Int -> [Int] -> [Int]
+rotate _ [] = []
+rotate a x = drop a (take g (cycle x))
+  where g = length x + a
```

Comment:
On negative `a` this won't work as expected.
Other than that, very elegant code 👍🏼